modules/mpi/tutorials/ospMPIDistribTutorialVolumeHDF5.cpp

```diff
@@ -90,13 +90,61 @@ int main(int argc, char **argv)
         },
         nullptr);
 
+    std::cerr << "ABOUT TO READ" << std::endl;
+
     // open the hdf5 file, read the data and create the ospray Volume object
     ChomboHDF5::Reader hdf5reader(argv[1], mpiRank, mpiWorldSize, argv[2]);
+    std::cerr << "READ!" << std::endl;
+
+    VolumeBrick brick;
+
+    //convert from int to float as required by ospray
+    //get the world bounds while we are at it
+    std::vector<box3f> myregions;
+    vec3f min{MAXFLOAT};
+    vec3f max{-MAXFLOAT};
+    for (auto x : hdf5reader.m_blockBounds) {
+        std::cerr << x.lower << " " << x.upper << std::endl;
+        if (x.lower.x < min.x)
+            min.x = x.lower.x;
+        if (x.lower.y < min.y)
+            min.y = x.lower.y;
+        if (x.lower.z < min.z)
+            min.z = x.lower.z;
+        if (x.lower.x > max.x)
+            max.x = x.lower.x;
+        if (x.lower.y > max.y)
+            max.y = x.lower.y;
+        if (x.lower.z > max.z)
+            max.z = x.lower.z;
+        myregions.push_back(box3f(x.lower, x.upper));
```

Comment:
I think this is wrong as all ranks store all block bounds. This should probably be conditioned on `ChomboHDF5::Reader::m_rankDataOwner == rank` or something.

Reply:
I have added a function: `ChomboHDF5::Reader::getMyRegions` which should return the object you need for myregions (i.e. only the bounds of the boxes on this rank).

Reply:
updated the viewer commit to use getMyRegions. ty